### PR TITLE
Add some helpers functions for testing purposes.

### PR DESCRIPTION
**Pull request by smallhive**

### CODE REVIEW

File: contracts/deploys.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package contracts
```
Comment (roman-khimov) [inline contracts/deploys.go:1]: Wrong place for this.

File: contracts/deploys.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import (
+	"testing"
```
Comment (roman-khimov) [inline contracts/deploys.go:4]: The package is for production apps, while we get `testing`, `neotest` and other things here.

File: contracts/deploys.go (hunk 1 above)
Comment (smallhive) [reply contracts/deploys.go]: Moved the package to `tests`

File: contracts/deploys.go (hunk 2 above)
Comment (smallhive) [reply contracts/deploys.go]: Moved

File: tests/deploys.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		ContractHash: contractHash,
+		Email:        "ops@nspcc.ru",
+		Refresh:      3600,
+		Retry:        600,
+		Expire:       int64(365 * 24 * time.Hour / time.Millisecond), // 1 year.
+		TTL:          3600,
+	}
+}
+
+// RegContractNNS make registration of the contract hash in nns contract.
```
Comment (cthulhu-rider) [inline tests/deploys.go:41]: proposed name is about NNS contact itself to me, i suggest`RegisterContractName` or `RegisterContractInNNS`?

File: tests/deploys.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
+	require.NoError(t, err)
+
+	nnsInv := e.CommitteeInvoker(nnsHash)
+
+	nnsInv.Invoke(t, true, "register", params.Name+".neofs", e.CommitteeHash, params.Email, params.Refresh, params.Retry, params.Email, params.TTL)
+	nnsInv.Invoke(t, nil, "addRecord", params.Name+".neofs", 16, params.ContractHash.StringLE())
+}
+
+// TickEpoch increase epoch value by one.
```
Comment (cthulhu-rider) [inline tests/deploys.go:52]:
```suggestion
// TickEpoch increases epoch value by one.
```
or just `increments`

File: tests/deploys.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+func TickEpoch(t *testing.T, e *neotest.Executor, signers ...neotest.Signer) {
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
+	require.NoError(t, err)
+
+	nnsInv := e.CommitteeInvoker(nnsHash)
+	resolveResult, err := nnsInv.TestInvoke(t, "resolve", "netmap.neofs", 16)
```
Comment (cthulhu-rider) [inline tests/deploys.go:58]: is there any way to simplify implementation? We have actors and contract-based wrappers, but still do some kind of surgery. I expect `neotest.Executor` to fit interface requirements

**Review by cthulhu-rider — COMMENTED** [review]
helpers should be tested themselves, especially with current implementation

File: tests/deploys.go (hunk 3 above)
Comment (smallhive) [reply tests/deploys.go]: Renamed to `RegisterContractInNNS`

File: tests/deploys.go (hunk 4 above)
Comment (smallhive) [reply tests/deploys.go]: Updated

File: tests/deploys.go (hunk 5 above)
Comment (smallhive) [reply tests/deploys.go:58]: Neotest functionality is quite poor and low level. I failed to find a simpler way as well

**Comment by smallhive** [comment]
> helpers should be tested themselves, especially with current implementation

I added tests

File: tests/deploys.go (hunk 5 above)
Comment (roman-khimov) [reply tests/deploys.go:58]: `neotest` isn't very friendly to invokers/actors now (https://github.com/nspcc-dev/neo-go/issues/3245).

File: tests/deploys_test.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+			nnsContract := neotest.Contract{
+				Hash:     state.CreateContractHash(e.CommitteeHash, fsContract.NEF.Checksum, fsContract.Manifest.Name),
+				NEF:      &fsContract.NEF,
+				Manifest: &fsContract.Manifest,
+			}
+
+			e.DeployContract(t, &nnsContract, []any{[]any{[]any{"neofs", "ops@nspcc.io"}}})
+		}
+
+		if fsContract.Manifest.Name == "NeoFS Netmap" {
```
Comment (cthulhu-rider) [inline tests/deploys_test.go:31]: `switch` with previous case

File: tests/deploys_test.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+				false, // notaryDisabled is false
+				nil,
+				nil,
+				nil,
+				[]any{},
+			}
+
+			e.DeployContract(t, &netMapContract, netMapContractDeployData)
+
+			params := NewRegisterContractInNNSParams("netmap", netMapContract.Hash)
```
Comment (cthulhu-rider) [inline tests/deploys_test.go:48]: following steps should be done outside the loop

File: tests/deploys_test.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+			RegisterContractInNNS(t, e, params)
+			require.NoError(t, err)
```
Comment (cthulhu-rider) [inline tests/deploys_test.go:50]: check that contract can be resolved is missing

File: tests/deploys_test.go (hunk 6 above)
Comment (smallhive) [reply tests/deploys_test.go]: Updated

File: tests/deploys_test.go (hunk 7 above)
Comment (smallhive) [reply tests/deploys_test.go]: Reworked the code a bit

**Review by cthulhu-rider — DISMISSED** [review]

File: tests/deploys.go [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+// RegisterContractInNNS make registration of the contract hash in nns contract.
+func RegisterContractInNNS(t *testing.T, e *neotest.Executor, params RegisterContractInNNSParams) {
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
+	require.NoError(t, err)
+
+	nnsInv := e.CommitteeInvoker(nnsHash)
+
+	nnsInv.Invoke(t, true, "register", params.Name+".neofs", e.CommitteeHash, params.Email, params.Refresh, params.Retry, params.Email, params.TTL)
+	nnsInv.Invoke(t, nil, "addRecord", params.Name+".neofs", 16, params.ContractHash.StringLE())
```
Comment (carpawell) [inline tests/deploys.go:70]: can we use https://github.com/nspcc-dev/neofs-contract/blob/db6dbb9fa0116f95b7532c84266c4ca84cb94227/contracts/nns/recordtype/recordtype.go#L15?

File: tests/deploys.go [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+		Email:        "ops@nspcc.ru",
+		Refresh:      3600,
+		Retry:        600,
+		Expire:       int64(365 * 24 * time.Hour / time.Millisecond), // 1 year.
+		TTL:          3600,
+	}
+}
+
+func nnsContractInvoker(t *testing.T, e *neotest.Executor) *neotest.ContractInvoker {
+	nnsHash, err := e.Chain.GetContractScriptHash(1)
```
Comment (carpawell) [inline tests/deploys.go:42]: i think @roman-khimov prefers https://github.com/nspcc-dev/neofs-contract/blob/db6dbb9fa0116f95b7532c84266c4ca84cb94227/rpc/nns/hashes.go#L34 or at least const from this package

**Review by carpawell — COMMENTED** [review]
In general, isn't most of the code can (should) be reused from `rpc/nns` package? It seems like it was made for it but mb I don't know some interfaces mismatch.

File: tests/deploys.go (hunk 10 above)
Comment (smallhive) [reply tests/deploys.go:43]: So do I, but I have nothing to suggest as a `ContractStateGetter` for InferHash func.

File: tests/deploys.go (hunk 9 above)
Comment (smallhive) [reply tests/deploys.go]: Updated

File: tests/deploys.go [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+	"github.com/nspcc-dev/neo-go/pkg/neotest"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neofs-contract/rpc/nns"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	// RegisterContractInNNSParams is a param struct for RegisterContractInNNS func.
+	RegisterContractInNNSParams struct {
+		// Name is a contract name. Name must not be a top level domain. For instance: netmap.
```
Comment (roman-khimov) [inline tests/deploys.go:16]: It's not about TLD. It's about collisions with NeoFS contracts. See `rpc/nns/names.go`.

File: tests/deploys.go [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+		// The final registered domain name will be concatenated with `.neofs`.
```
Comment (roman-khimov) [inline tests/deploys.go:17]: Domain will be registered in the `.neofs` TLD.

File: tests/deploys.go (hunk 10 above)
Comment (roman-khimov) [reply tests/deploys.go:43]: nns.ID is still relevant here.

File: tests/deploys.go [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+func getContractHash(t *testing.T, e *neotest.Executor, name string) util.Uint160 {
+	nnsInv := nnsContractInvoker(t, e)
+	resolveResult, err := nnsInv.TestInvoke(t, "resolve", name, nns.TXT)
+	require.NoError(t, err)
+
+	contractHashBytes, err := resolveResult.Pop().Array()[0].TryBytes()
+	require.NoError(t, err)
+
+	netMapContractHash, err := util.Uint160DecodeStringLE(string(contractHashBytes))
```
Comment (roman-khimov) [inline tests/deploys.go:57]: At least reuse `nns.AddressFromRecord()` here.

File: tests/deploys.go (hunk 12 above)
Comment (smallhive) [reply tests/deploys.go]: Changed

File: tests/deploys.go (hunk 13 above)
Comment (smallhive) [reply tests/deploys.go]: Replaced

**Review by roman-khimov — APPROVED** [review]

File: tests/deploys.go (hunk 10 above)
Comment (carpawell) [reply tests/deploys.go]: > nothing to suggest as a ContractStateGetter for InferHash func

wasnt `e.Chain` good enough?

Comment (smallhive) [reply tests/deploys.go]: Unfortunately no, neotest Chain is not good enough

**Pull request merged**